### DISCUSSION TITLE
ZbOccupancy vs ZoneStatusChange (PIR ZIGBEE SENSOR)

### DISCUSSION THREAD

**Discussion by ortegafernando** [arendst/Tasmota#12759] [Q&A] [answered]
Hi, thanks a lot for this great "firmware".
I have working perfectly a sonoff bridge with last version of tasmota.
I have a "problem" with ZoneStatusChange and zboccupancy.

I think it is not an issue, but it may be.

ZbOccupancy is not working as expected. I mean: once I walk in front of it (ONLY ONE TIME), ZbOccupancy says "1", and after 90 seconds (default time out) it sends "0".
ZoneStatusChange shows the same way.

But if I keep walking in front of it during 3 minutes, for example, ZoneStatusChange sends "1" at the beggining and, after a timeout that i am not sure how much it is, sends "0".

But ZbOccupancy says "1" at the begginig, and after 90 seconds, sends "0", but I keep walking in fron of it !!!!

Then... what is the real porpuse of ZbOccupancy?

Here you have data:
left panel: ZoneStatusChange. 09:04 walk only one time, and almost at 09:11 walk for about 3 minutes
right panel: ZbOccupancy.

Do you see the problem? Have you got the same problem?

Thanks.

![image](https://user-images.githubusercontent.com/8700791/126955001-b24314b6-2f83-4f5d-844d-7293bf117d60.png)

**Comment by s-hadinger** [comment, marked as answer]
It all depends on the device manufacturer. Some devices like Xiaomi only send "1" and an absence of "1" for an extended period of time must be understood as no occupancy. This is why Z2T has a default time-out of 90 seconds to generate a synthetic event "0".

Some other devices like Philips PIR send only one "1" event, and then a "0" event when there is no more occupancy. In such case you should disable the Z2T time-out.

See https://tasmota.github.io/docs/Commands/#zboccupancy

**Reply by ortegafernando** [reply]
Thanks a lot. Is disable this?

```
0: no time-out, the device actually generates "Occupancy":0
```

I understood just other way round. Sorry.

I will try, thanks, and tomorrow I will publish again graphs.

**Reply by ortegafernando** [reply]
Thanks @s-hadinger now, ZoneStatusChange and zbOccupancy seems to be the "same".

I attach here the new screenshoot with the same test as yesterday.

![image](https://user-images.githubusercontent.com/8700791/127268061-f45a6e91-e24a-4ea3-a5ac-9f4201bdce52.png)

Thanks a lot.

**Comment by barcelona92** [comment]
How too reduce time? 
Ihave same problem , the occupancy is more than 90 seconds

**Reply by s-hadinger** [reply]
Can you be more specific?

**Comment by barcelona92** [comment]
From the time it detects, until it stops detecting, 90 seconds pass, so if you happen again in less than 90 seconds it stays the same. It does not make it very functional, and that I have modified with the bridge.
